File: client/src/core/enums/enums.ts
export enum EventTier {
    TIER_0,
    TIER_1,
    TIER_2,
    TIER_3,
    TIER_4
}

export enum ChallengeCategoryMajor {
    CHALLENGE = 'Challenge',
    ZONE = 'Zone',
    COURSE = 'Course',
    SPRINT = 'Sprint',
    MILESTONE = 'Milestone',
    COLLECTABLE = 'Collectable',
    ACHIEVEMENT = 'Achievement'
}

export enum CourseCategoryMinor {
    STANDARD = 'Standard',
    LOOP = 'Loop',
    OPEN_GUIDED = 'Open Guided',
    OPEN_UNGUIDED = 'Open Unguided'
}

export enum SprintCategoryMinor {
    STANDARD = 'Standard'
}

export enum MilestoneCategoryMinor {
    DAILY = 'Daily',
    WEEKLY = 'Weekly',
    MONTHLY = 'Monthly'
}

export enum CompleteStatus {
    NOT_COMPLETE,
    STANDARD_T0,
    BRONZE_T1,
    SILVER_T2,
    GOLD_T3
}

// TODO: STILL needs fine tuning
export enum AssetType {
    // Items imported by enum ItemType
    ITEM = 'items',
    
    // Below types imported by id number
    BANNER = 'banners',
    ACHIEVEMENT = 'achievements',
    MILESTONE = 'milestones'
}

File: topchallenger_web_api/schema/enums.py
from enum import Enum

class ChallengeCategoryMajor(Enum):
    CHALLENGE = 'Challenge'
    ZONE = 'Zone'
    COURSE = 'Course'
    SPRINT = 'Sprint'
    MILESTONE = 'Milestone'
    COLLECTABLE = 'Collectable'
    ACHIEVEMENT = 'Achievement'

class CourseCategoryMinor(Enum):
    STANDARD = 0
    LOOP = 1
    OPEN_GUIDED = 2
    OPEN_UNGUIDED = 3


class SprintCategoryMinor(Enum):
    STANDARD = 0


class MilestoneCategoryMinor(Enum):
    DAILY = 'Daily'
    WEEKLY = 'Weekly'
    MONTHLY = 'Monthly'


class CompleteStatus(Enum):
    NOT_COMPLETE = 0
    STANDARD_T0 = 1
    BRONZE_T1 = 2
    SILVER_T2 = 3
    GOLD_T3 = 4
